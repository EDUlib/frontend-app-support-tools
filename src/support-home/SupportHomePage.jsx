import React from 'react';
import { Link } from 'react-router-dom';

const SupportHomePage = () => (
  <main className="container-fluid mt-3 mb-5">
    <h3>Support Tools</h3>
    <ul>
      <li>
        <Link id="search-users" to="/users">
          Search Users
        </Link>
      </li>
    </ul>
  </main>
);

export default SupportHomePage;
