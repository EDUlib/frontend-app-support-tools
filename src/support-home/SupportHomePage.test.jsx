import { shallow } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SupportHomePage from './SupportHomePage';

const SupportHomePageWrapper = () => (
  <MemoryRouter>
    <SupportHomePage />
  </MemoryRouter>
);

describe('SupportLinksPage', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SupportHomePage />);
  });

  it('correctly renders support links', () => {
    expect(wrapper.find('li')).toHaveLength(1);
  });
  it('has search users link', () => {
    const searchUsersText = wrapper.find('#search-users').text();

    expect(searchUsersText).toEqual('Search Users');
  });
  it('has correct page heading', () => {
    const supportToolsText = wrapper.find('h3').text();

    expect(supportToolsText).toEqual('Support Tools');
  });
  it('matches snapshot', () => {
    const tree = renderer.create(<SupportHomePageWrapper />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
