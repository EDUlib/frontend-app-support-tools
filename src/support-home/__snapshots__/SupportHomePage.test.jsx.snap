// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupportLinksPage matches snapshot 1`] = `
<main
  className="container-fluid m-5"
>
  <h3>
    Support Tools
  </h3>
  <ul>
    <li>
      <a
        href="/users"
        id="search-users"
        onClick={[Function]}
      >
        Search Users
      </a>
    </li>
  </ul>
</main>
`;
